src/types.rs

```diff
@@ -150,32 +150,132 @@ encode_decode_as!(Limits, {
     (Limits { min, max: Some(max) }) <=> (LimitsRepr::MinMax { min, max }),
 });
 
-#[cfg(feature = "threads")]
+#[cfg(any(feature = "threads", feature = "custom-page-sizes"))]
 #[derive(Wasmbin)]
 #[repr(u8)]
 enum MemTypeRepr {
     Unshared(LimitsRepr),
-    SharedMin { min: u32 } = 0x02,
-    SharedMinMax { min: u32, max: u32 } = 0x03,
+    #[cfg(feature = "threads")]
+    SharedMin {
+        min: u32,
+    } = 0x02,
+    #[cfg(feature = "threads")]
+    SharedMinMax {
+        min: u32,
+        max: u32,
+    } = 0x03,
+    #[cfg(feature = "custom-page-sizes")]
+    UnsharedMinCustom {
+        min: u32,
+        page_size: PageSize,
+    } = 0x08,
+    #[cfg(feature = "custom-page-sizes")]
+    UnsharedMinMaxCustom {
+        min: u32,
+        max: u32,
+        page_size: PageSize,
+    } = 0x09,
+    #[cfg(all(feature = "threads", feature = "custom-page-sizes"))]
+    SharedMinCustom {
+        min: u32,
+        page_size: PageSize,
+    } = 0x0A,
+    #[cfg(all(feature = "threads", feature = "custom-page-sizes"))]
+    SharedMinMaxCustom {
+        min: u32,
+        max: u32,
+        page_size: PageSize,
+    } = 0x0B,
+}
+
+#[cfg(feature = "custom-page-sizes")]
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Visit)]
+pub struct PageSize(u32);
+
+#[cfg(feature = "custom-page-sizes")]
+impl PageSize {
```

Comment:
Just noticed like there is currently no way to use the parsed / stored value.

Can you please add `pub const fn size_log2()` (to return raw stored value) and `pub const fn size()` (for human-readable value)?